### ISSUE TITLE
Pydantic model default const parameters (pydantic.Field) are not respected at construction time

### ISSUE THREAD

**Issue by SebastianMorawiec** [microsoft/pyright#6915]
Given the following code:

```python
from pydantic import BaseModel
import pydantic
from dataclasses import dataclass


class MyModel(BaseModel):
    const: str = pydantic.Field("my_constant", const=True)


@dataclass(frozen=True)
class A:
    x: int = 5


model = MyModel(id=5)

print(model.const)

a = A()
```
Default value for dataclass (class A) is respected, but default value for this BaseModel is not respected, and produces missing parameter error at construction time. The issue goes away if I dont use pydantic.Field as default value, but an actual string.

I am sorry if this is pydantic issue or it was already reported, but could not find anything similar. I only found this: https://github.com/microsoft/pyright/issues/6657 bug which is related but it does not actually describes issues with pydantic.Field as default being ignored.

This is reproduced using:
pydantic version 1.10.13
pyright version 1.1.344 (pyright installed as pip package)

**Comment by SebastianMorawiec** [comment]
I just found out that using
const: str = pydantic.Field(default="my_constant", const=True)
instead of 
const: str = pydantic.Field("my_constant", const=True)

fixed the warning, but the original report still seems puzzling to me

**Comment by Viicos** [comment]
`default` should be used as a keyword argument so that type checkers can accurately provide a signature. See my answer here: https://github.com/pydantic/pydantic/discussions/7379.

**Comment by erictraut** [comment]
Pyright has no special knowledge of pydantic or its semantics. Pyright uses the Python typing standard and the type information it's provided in libraries and stub files.

Pydantic makes use of the `dataclass_transform` mechanism introduced in [PEP 681](https://peps.python.org/pep-0681/). This mechanism allows a library like pydantic to indicate to type checkers that they behave like the stdlib `dataclass` mechanism. Pydantic diverges from the standard dataclass in a few small ways, and pyright doesn't have any knowledge of these non-standard behaviors. One example is the ability to pass a default via a positional argument rather than specifying it using a `default` keyword argument, as you discovered above. In other words, if you want your code to work with standards-compliant type checkers like pyright, you'll need to stick to pydantic features that are available in `dataclass`.
